core/src/main/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingService.scala

```diff
@@ -470,32 +450,85 @@ class SpendReportingService(
     bytesProcessedCounter += stats.getEstimatedBytesProcessed
   }
 
+  def getSpendReportableWorkspaceGoogleProjects(
+    childContext: RawlsRequestContext
+  ): Future[Map[RawlsBillingProjectName, Seq[Workspace]]] =
+    samDAO
+      .listResourcesWithActions(
+        SamResourceTypeNames.workspace,
+        SamWorkspaceActions.readSpendReport,
+        childContext
+      )
+      .flatMap {
+        case ownerWorkspaces if ownerWorkspaces.isEmpty =>
+          Future.successful(Map.empty[RawlsBillingProjectName, Seq[Workspace]])
```

Comment:
Why is this case necessary? Will `getGCPWorkspacesByBillingProjects` fail if `validWorkspaceIds` is empty?

Reply:
Fixed: https://github.com/broadinstitute/rawls/pull/3193/commits/6ca541dbe6e14fd161620ceb45f298cda6aabc57